packages/expo-firebase-analytics/src/Analytics.ts

```diff
@@ -0,0 +1,198 @@
+ *     to clear the current screen name.
+ * @param screenClassOverride The name of the screen class. Should contain 1 to 100 characters. By
+ *     default this is the class name of the current screen (UIViewController on iOS). Set to `undefined` to revert to the
+ *     default class name.
+ */
+export async function setCurrentScreen(
+  screenName?: string,
+  screenClassOverride?: string
+): Promise<void> {
+  if (!ExpoFirebaseAnalytics.setCurrentScreen) {
+    throw new UnavailabilityError('expo-firebase-analytics', 'setCurrentScreen');
+  }
+  return await ExpoFirebaseAnalytics.setCurrentScreen(screenName, screenClassOverride);
+}
+
+/**
+ * Sets the interval of inactivity in seconds that terminates the current session. The default
+ * value is 1800000 milliseconds (30 minutes).
+ *
+ * @param sessionTimeoutInterval The custom time of inactivity in milliseconds before the current
+ *     session terminates.
+ */
+export async function setSessionTimeoutDuration(sessionTimeoutInterval: number): Promise<void> {
+  if (!ExpoFirebaseAnalytics.setSessionTimeoutDuration) {
+    throw new UnavailabilityError('expo-firebase-analytics', 'setSessionTimeoutDuration');
+  }
+  return await ExpoFirebaseAnalytics.setSessionTimeoutDuration(sessionTimeoutInterval);
+}
+
+/**
+ * Sets the user ID property. This feature must be used in accordance with
+ * [Google's Privacy Policy](https://www.google.com/policies/privacy)
+ *
+ * @param userID The user ID to ascribe to the user of this app on this device, which must be
+ *     non-empty and no more than 256 characters long. Setting userID to null removes the user ID.
+ */
+export async function setUserId(userId: string | null): Promise<void> {
+  if (!ExpoFirebaseAnalytics.setUserId) {
+    throw new UnavailabilityError('expo-firebase-analytics', 'setUserId');
+  }
+  return await ExpoFirebaseAnalytics.setUserId(userId);
+}
+/**
+ * Sets a user property to a given value. Up to 25 user property names are supported. Once set,
+ * user property values persist throughout the app life-cycle and across sessions.
+ *
+ * The following user property names are reserved and cannot be used:
+ *
+ * - `first_open_time`
+ * - `last_deep_link_referrer`
+ * - `user_id`
+ *
+ * @param name The name of the user property to set. Should contain 1 to 24 alphanumeric characters
+ *     or underscores and must start with an alphabetic character. The "firebase_", "google_", and
+ *     "ga_" prefixes are reserved and should not be used for user property names.
+ * @param value The value of the user property. Values can be up to 36 characters long. Setting the
+ *     value to null removes the user property.
+ */
+export async function setUserProperty(name: string, value: string): Promise<void> {
+  if (!ExpoFirebaseAnalytics.setUserProperty) {
+    throw new UnavailabilityError('expo-firebase-analytics', 'setUserProperty');
+  }
+  return await ExpoFirebaseAnalytics.setUserProperty(name, value);
+}
```

Comment:
`setUserProperty` is a method only provided by the native SDKs, while the JS SDK follows the `setUserProperties` method. The question here is, which API should we follow here? I would opt for following the JS API design here and only provide native methods when they add any additional value. The additional value for having `setUserProperty` here could be that a user can do fine-grained checks on whether each user-property set action succeeded or failed, but that's really it.
Having a single `setUserProperties` function here would make the API footprint smaller and would provide also better performance. By hoisting all the properties over the bridge at once, the performance would be optimal and the native code can then easily use the single-property setter method to implement the behavior.